types/request.d.ts

```diff
@@ -64,7 +64,7 @@ export interface FastifyRequest<RouteGeneric extends RouteGenericInterface = Rou
   body: RequestType['body'];
   context: FastifyRequestContext<ContextConfig>;
   routeConfig: FastifyRequestContext<ContextConfig>['config'];
-  routeSchema: FastifySchema
+  routeSchema?: FastifySchema
```

Comment:
Not sure how to document it

```suggestion
  routeSchema?: FastifySchema // it is empty for 404 requests
```